kwave/kWaveSimulation.py

```diff
@@ -28,28 +28,48 @@ class kWaveSimulation(object):
 
     def __init__(self,
                  kgrid: kWaveGrid,
+                 source: kSource,
+                 sensor: NotATransducer,
                  medium: kWaveMedium,
-                 source,
-                 sensor,
-                 **kwargs
-                 ):
+                 simulation_options: SimulationOptions):
         self.precision = None
         self.kgrid = kgrid
         self.medium = medium
         self.source = source
         self.sensor = sensor
-        self.kwargs = kwargs
+        self.options = simulation_options
 
         # =========================================================================
         # FLAGS WHICH DEPEND ON USER INPUTS (THESE SHOULD NOT BE MODIFIED)
         # =========================================================================
         # flags which control the type of simulation
         #: Whether simulation type is axisymmetric
-        self.userarg_axisymmetric      = kwargs['axisymmetric']
+        self.userarg_axisymmetric      = self.options.axisymmetric
 
         # flags which control the characteristics of the sensor
         #: Whether time reversal simulation is enabled
-        self.userarg_time_rev          = kwargs['time_rev']
+
+        # check if performing time reversal, and replace inputs to explicitly use a
+        # source with a dirichlet boundary condition
+        if self.sensor.time_reversal_boundary_data is not None:
```

Comment:
could this be added to simulation options?

Reply:
Is this field related to overall simulation? If not, I'd keep it here. If yes, I'd still keep it here because it is not used/tested in the code. We cannot validate if the functionality/behavior is preserved after the refactoring.